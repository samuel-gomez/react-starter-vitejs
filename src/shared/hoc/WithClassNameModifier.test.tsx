import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import withClassNameModifier from './WithClassNameModifier';

const CustomComponent = ({ className }: { className?: string }) => <div className={className}>Hello</div>;

describe('WithClassNameModifier', () => {
  const EnhancedComponent = withClassNameModifier(CustomComponent);
  it('Should render Custom Component with new className props when apply HOC with className', () => {
    const { asFragment } = render(<EnhancedComponent className="af-component" classModifier="sam" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render Custom Component with new className props with 2 modifiers when apply HOC with className with 2 modifiers', () => {
    const { asFragment } = render(<EnhancedComponent className="af-component" classModifier="sam sam2" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render Custom Component with new className props null modifier when apply HOC with className null modifier', () => {
    const { asFragment } = render(<EnhancedComponent className="af-component" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render Custom Component without className props when apply HOC with className null modifier', () => {
    const { asFragment } = render(<EnhancedComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
