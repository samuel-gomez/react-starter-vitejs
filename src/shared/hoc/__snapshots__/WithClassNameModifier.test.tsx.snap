// Vitest Snapshot v1

exports[`WithClassNameModifier > Should render Custom Component with new className props null modifier when apply HOC with className null modifier 1`] = `
<DocumentFragment>
  <div
    class="af-component"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`WithClassNameModifier > Should render Custom Component with new className props when apply HOC with className 1`] = `
<DocumentFragment>
  <div
    class="af-component af-component--sam"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`WithClassNameModifier > Should render Custom Component with new className props with 2 modifiers when apply HOC with className with 2 modifiers 1`] = `
<DocumentFragment>
  <div
    class="af-component af-component--sam af-component--sam2"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`WithClassNameModifier > Should render Custom Component without className props when apply HOC with className null modifier 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    Hello
  </div>
</DocumentFragment>
`;
